rectools/metrics/classification.py

```diff
@@ -85,6 +90,9 @@ def calc_per_user(self, reco: pd.DataFrame, interactions: pd.DataFrame, catalog:
         pd.Series
             Values of metric (index - user id, values - metric value for every user).
         """
+        if self.debias_config is not None:
```

Comment:
I think it's really unclear and unsafe that  we execute downsampling in `calc_per_user`, but not in `calc_per_user_from_confusion_df`. 
User would probably like to use `calc_per_user_from_confusion_df` themselves but they wouldn't expect debias is not executed there

Reply:
Sounds reasonable, but you will need to call this private method from `calc_classification_metrics`, right? Not the best practice

Reply:
From `calc_classification_metrics`, yes. What do you suggest?
Another option is to make `calc_per_user_from_confusion_df` public and write a docstring. I don't see other ways.

Reply:
Or we can add boolean argument, smth like "is_confusion_df_debiased".
If it is None or False -> Check is debias_config is not None -> through error

Reply:
Right

Reply:
And the method remains public. So this is actually a breaking change and we need to write it in the changelog.

Reply:
Why is it a breaking change? If you don't specify the debias_config, your old code will continue to work, no?

Reply:
We're adding a required (not optional) argument to a public method. So old code will not work if someone actually used this method.
To make it non-breaking we need this argument to be optional and treat `None` as `False`

Reply:
The argument is not optional, but it has a default value. The old code should work :)

Reply:
You are right. My bad :)

Reply:
Am I understanding you correctly that we come to the point where we add an argument `is_confusion_df_debiased` to `ClassificationMetric`, `ClassificationMetric`? 
I added `is_confusion_df_debiased` as argument, because the `calc` method reaches the `calc_per_user_from_confusion_df` method, where this argument is needed. But if you don't pass it to `calc` and use debias mechanism, you will get an error with this [algorithm](https://github.com/MobileTeleSystems/RecTools/pull/152#discussion_r1633793551)

Reply:
We need `is_confusion_df_debiased` as an argument to `calc_per_user_from_confusion_df` method. Not as an argument to `ClassificationMetric` class.
We set it to `False` by default.
If we did debiasing in the code then we call `calc_per_user_from_confusion_df` method with `is_confusion_df_debiased` set to True.

Reply:
Yes, it's correct now, thank you!
@feldlime will resolve since it is his thread